DeepSpeech.py

```diff
@@ -694,594 +651,999 @@ def stopwatch(start_duration=0):
-
-        # Compute the batch
-        result = session.run(params, **extra_params)
-
-        # Add batch to loss
-        total_loss += result[param_idx['avg_loss']]
-
-        if do_training:
-            # Log all variable states in current step
-            step = epoch * data_set.total_batches + batch * len(available_devices)
-            if log_variables:
-                writer.add_summary(result[param_idx['merged']], step)
-            if do_fulltrace:
-                writer.add_run_metadata(loss_run_metadata, 'loss_epoch%d_batch%d'   % (epoch, batch))
-            writer.flush()
-
-        if query_report:
-            sample_results = result[param_idx['sample_results']]
-            # Collect individual sample results
-            collect_results(report_results, sample_results[0])
-            # Add batch to total_accuracy
-            total_accuracy += sample_results[1]
-
-    # Returning the batch set result tuple
-    loss = total_loss / batches_per_device
-    if query_report:
-        return (loss, total_accuracy / batches_per_device, report_results)
-    else:
-        return (loss, None, None)
-
-
-def print_report(caption, batch_set_result):
-    r"""
-    This routine will print a report from a provided batch set result tuple.
-    It takes a caption for titling the output plus the batch set result tuple.
-    If the batch set result tuple contains accuracy and a report results tuple,
-    a complete WER report will be calculated, printed and its mean WER returned.
-    Otherwise it will just print the loss and return ``None``.
-    """
-    # Unpacking batch set result tuple
-    loss, accuracy, results_tuple = batch_set_result
-
-    # We always have a loss value
-    title = caption + " loss=" + "{:.9f}".format(loss)
-
-    mean_wer = None
-    if accuracy is not None and results_tuple is not None:
-        title += " avg_cer=" + "{:.9f}".format(accuracy)
-        mean_wer = calculate_and_print_wer_report(title, results_tuple)
-    else:
-        print(title)
-
-    return mean_wer
+            return 'Test%s' % ename
+
+    def get_job(self, worker):
+        '''Gets the next open job from this epoch. The job will be marked as 'running'.
+
+        Args:
+            worker (int): index of the worker that takes the job
+
+        Returns:
+            WorkerJob. job that has been marked as running for this worker
+        '''
+        if len(self.jobs_open) > 0:
+            job = self.jobs_open.pop(0)
+            self.jobs_running.append(job)
+            job.worker = worker
+            return job
+        else:
+            return None
+
+    def finish_job(self, job):
+        '''Finishes a running job. Removes running representant from the epoch
+        and adds the provided `job` as 'done' to the epoch.
+
+        Args:
+            job (WorkerJob): the job to put into state 'done'
+        '''
+        index = next((i for i in xrange(len(self.jobs_running)) if self.jobs_running[i].id == job.id), -1)
```

Comment:
`WorkerJob` has a unique id, so why not make `jobs_open/running/one` dicts? Then you can just do:
```python
    del self.jobs_running[job.id]
    self.jobs_done[job.id] = job
```